### PR TITLE
Support the -Wassign-enum compiler flag.

### PR DESCRIPTION
**Pull request by sergiocampama**

### CODE REVIEW

**Comment by sergiocampama** [comment]
fyi @thomasvl

File: objectivec/GPBDescriptor_PackagePrivate.h [hunk 1]
```diff
@@ -37,6 +37,7 @@
 
 // Describes attributes of the field.
 typedef NS_OPTIONS(uint16_t, GPBFieldFlags) {
+  GPBFieldFlagNone        = 0,
```
Comment (thomasvl) [inline objectivec/GPBDescriptor_PackagePrivate.h]: GPBFieldNone to make the style of these other ones.

File: objectivec/GPBDescriptor_PackagePrivate.h [hunk 2]
```diff
@@ -111,6 +112,7 @@ typedef struct GPBMessageFieldDescriptionWithDefault {
 
 // Describes attributes of the extension.
 typedef NS_OPTIONS(uint8_t, GPBExtensionOptions) {
+  GPBExtensionOptionNone    = 0,
```
Comment (thomasvl) [inline objectivec/GPBDescriptor_PackagePrivate.h]: GPBExtensionNone 

also to match

File: src/google/protobuf/compiler/objectivec/objectivec_helpers.cc [hunk 3]
```diff
@@ -621,6 +621,33 @@ ObjectiveCType GetObjectiveCType(FieldDescriptor::Type field_type) {
...
+string GetZeroEnumNameForFlagType(const FlagType flag_type) {
+  switch(flag_type) {
+    case FLAGTYPE_DESCRIPTOR_INITIALIZATION:
+      return "GPBDescriptorInitializationFlag_None";
+    case FLAGTYPE_EXTENSION:
+      return "GPBExtensionOptionNone";
+    case FLAGTYPE_FIELD:
+      return "GPBFieldFlagNone";
+    default:
+      return "0";
```
Comment (thomasvl) [inline src/google/protobuf/compiler/objectivec/objectivec_helpers.cc]: GOOGLE_LOG(FATAL) << "Can't get here.";
  return "0";

File: src/google/protobuf/compiler/objectivec/objectivec_helpers.cc [hunk 4]
```diff
@@ -621,6 +621,33 @@ ObjectiveCType GetObjectiveCType(FieldDescriptor::Type field_type) {
...
+string GetEnumNameForFlagType(const FlagType flag_type) {
+  switch(flag_type) {
+    case FLAGTYPE_DESCRIPTOR_INITIALIZATION:
+      return "GPBDescriptorInitializationFlags";
+    case FLAGTYPE_EXTENSION:
+      return "GPBExtensionOptions";
+    case FLAGTYPE_FIELD:
+      return "GPBFieldFlags";
+    default:
+      // Unknown flag type.
```
Comment (thomasvl) [inline src/google/protobuf/compiler/objectivec/objectivec_helpers.cc]: GOOGLE_LOG(FATAL) << "Can't get here.";
return "";

File: src/google/protobuf/compiler/objectivec/objectivec_helpers.cc [hunk 5]
```diff
@@ -621,6 +621,33 @@ ObjectiveCType GetObjectiveCType(FieldDescriptor::Type field_type) {
   return OBJECTIVECTYPE_INT32;
 }
 
+string GetZeroEnumNameForFlagType(const FlagType flag_type) {
```
Comment (thomasvl) [inline src/google/protobuf/compiler/objectivec/objectivec_helpers.cc]: Move these two up into the anon namespace (namespace { ... }), so they are private to this file and don't become global symbols.

File: objectivec/GPBDescriptor_PackagePrivate.h (hunk 1 above)
Comment (sergiocampama) [inline objectivec/GPBDescriptor_PackagePrivate.h]: Done

File: objectivec/GPBDescriptor_PackagePrivate.h (hunk 2 above)
Comment (sergiocampama) [inline objectivec/GPBDescriptor_PackagePrivate.h]: Done

File: src/google/protobuf/compiler/objectivec/objectivec_helpers.cc (hunk 3 above)
Comment (sergiocampama) [inline src/google/protobuf/compiler/objectivec/objectivec_helpers.cc]: Done

File: src/google/protobuf/compiler/objectivec/objectivec_helpers.cc (hunk 4 above)
Comment (sergiocampama) [inline src/google/protobuf/compiler/objectivec/objectivec_helpers.cc]: Done

File: src/google/protobuf/compiler/objectivec/objectivec_helpers.cc (hunk 5 above)
Comment (sergiocampama) [inline src/google/protobuf/compiler/objectivec/objectivec_helpers.cc]: Done

File: objectivec/GPBDescriptor_PackagePrivate.h [hunk 6]
```diff
@@ -37,6 +37,7 @@
 
 // Describes attributes of the field.
 typedef NS_OPTIONS(uint16_t, GPBFieldFlags) {
+  GPBFieldNone        = 0,
```
Comment (thomasvl) [inline objectivec/GPBDescriptor_PackagePrivate.h]: Looks like the =0 indents need tweaking on two of these.

Comment (sergiocampama) [inline objectivec/GPBDescriptor_PackagePrivate.h]: argh... done :)

File: objectivec/GPBArray.m [hunk 7]
```diff
@@ -406,7 +406,7 @@ - (NSString *)description {
 }
 
 - (void)enumerateValuesWithBlock:(void (^)(int32_t value, NSUInteger idx, BOOL *stop))block {
-  [self enumerateValuesWithOptions:0 usingBlock:block];
+  [self enumerateValuesWithOptions:(NSEnumerationOptions)0 usingBlock:block];
```
Comment (thomasvl) [inline objectivec/GPBArray.m]: Oops, this needs to be updated in the template part of the file also.  That's what the travis failure is catching.

**Pull request merged**